metadata/src/main/java/org/apache/kafka/metadata/authorizer/ClusterMetadataAuthorizer.java

```diff
@@ -76,14 +78,15 @@ public interface ClusterMetadataAuthorizer extends Authorizer {
     void loadSnapshot(Map<Uuid, StandardAcl> acls);
 
     /**
-     * Add a new ACL. Any ACL with the same ID will be replaced.
-     */
-    void addAcl(Uuid id, StandardAcl acl);
-
-    /**
-     * Remove the ACL with the given ID.
+     * Add or remove ACLs.
+     *
+     * @param newAcls           The ACLs to add.
+     * @param removedAclIds     The ACL IDs to remove.
      */
-    void removeAcl(Uuid id);
+    void applyAclChanges(
```

Comment:
We should document that this method does not expect duplicates or allow replacing ACL by ID.